CRM/Membershipextrasimporterapi/EntityImporter/Contribution.php

```diff
@@ -149,8 +149,21 @@ private function getTotalAmount() {
   }
 
   private function getCurrency() {
-    // todo : not in mapping document, need to discuss with others how to get it.
-    return 'GBP';
+    if (!isset($this->cachedValues['currencies_enabled'])) {
+      $sqlQuery = "SELECT cov.name as name, cov.value as id FROM civicrm_option_value cov 
+                  INNER JOIN civicrm_option_group cog ON cov.option_group_id = cog.id 
+                  WHERE cog.name = 'currencies_enabled'";
+      $result = CRM_Core_DAO::executeQuery($sqlQuery);
+      while ($result->fetch()) {
+        $this->cachedValues['currencies_enabled'][$result->name] = $result->id;
+      }
+    }
+
+    if (!empty($this->cachedValues['currencies_enabled'][$this->rowData['contribution_currency']])) {
+      return $this->cachedValues['currencies_enabled'][$this->rowData['contribution_currency']];
+    }
+
+    throw new CRM_Membershipextrasimporterapi_Exception_InvalidContributionFieldException('Invalid or disabled contribution "currency"', 400);
```

Comment:
Just curious, do we have exception code document somewhere?

Reply:
Not really, but we are sending a message telling what the error is.

Reply:
I was thinking if the error was throwing the number, how would I know, but your reply gives me the answer already.